apps/self-hosted/hosting/scripts/hivesigner-redirect-uris.py

```
@@ -28,46 +38,83 @@
+# the broadcast.
+CONFIRM_ATTEMPTS = 8
+CONFIRM_DELAY_SECONDS = 3
+
+
+def log(message: str) -> None:
+    print(message, flush=True)
+
+
+def rpc(url: str, method: str, params: list) -> dict:
+    payload = json.dumps({"jsonrpc": "2.0", "method": method, "params": params, "id": 1}).encode()
+    request = urllib.request.Request(url, data=payload, headers={"Content-Type": "application/json"})
```

Ruff (0.16.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)